packages/native/src/NavigationContainer.tsx

```diff
@@ -46,13 +48,27 @@ const NavigationContainer = React.forwardRef(function NavigationContainer(
     ...linking,
   });
 
-  const [isReady, initialState] = useThenable(getInitialState);
+  const [isResolved, initialState] = useThenable(getInitialState);
 
   React.useImperativeHandle(ref, () => refContainer.current);
 
   const linkingContext = React.useMemo(() => ({ options: linking }), [linking]);
 
-  if (rest.initialState == null && isLinkingEnabled && !isReady) {
+  const isReady = rest.initialState != null || !isLinkingEnabled || isResolved;
+
+  const onReadyRef = React.useRef(onReady);
+
+  React.useEffect(() => {
+    onReadyRef.current = onReady;
+  });
```

Comment:
Without `useEffect` should be more stable:

```
const onReadyRef = React.useRef(onReady);

onReadyRef.current = onReady;
```

Reply:
That's not concurrent mode safe. Ref shouldn't be assigned during render.